### PR TITLE
HTML Notifications

### PR DESCRIPTION
**Pull request by ArdaXi**
This should work, I used code from Twitification (my own project). Had to include jQuery for it, but I have no idea if there's a better way to do this. Well, see what you can do with it. Great project!

### CODE REVIEW

**Comment by badboy** [comment]
There are already notifications implemented in the current code. What exactly is the advantage of your way? (I didn't test it and I just had a quick look on the source code)

Btw.: jquery is already included using the google cdn

**Comment by ArdaXi** [comment]
This way you get HTML support in the notifications, which means you can click the name of the author to go to their Twitter page, click URLs in the tweet, et cetera. The existing approach is very limited because it's text-only.

File: public/ext/jquery-1.4.2.min.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
+/*!
```
Comment (cramforce) [inline public/ext/jquery-1.4.2.min.js]: jQuery is loaded from the CDN already

File: public/ext/jquery.timeago.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
+/*
```
Comment (cramforce) [inline public/ext/jquery.timeago.js:1]: While this is probably much better than what is in streamie, would be possible to use the age-functionality in streamie? Or substitute is fully with this?

File: public/ext/notification.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
+
```
Comment (cramforce) [inline public/ext/notification.js]: Please move the file to its own subdirectory

File: public/ext/notification.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  return date.getUTCFullYear()
+    + "-" + zeropad(date.getUTCMonth()+1)
+    + "-" + zeropad(date.getUTCDate())
+    + "T" + zeropad(date.getUTCHours())
+    + ":" + zeropad(date.getUTCMinutes())
+    + ":" + zeropad(date.getUTCSeconds()) + "Z";
+};
+
+function escapeHTML(text) {
+    var div = document.createElement("div");
```
Comment (cramforce) [inline public/ext/notification.js]: indent

File: public/ext/notification.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	div.innerText = text;
+	return div.innerHTML;
+}
+
+function linkify_entities(tweet) {
```
Comment (cramforce) [inline public/ext/notification.js]: Why not use the HTML formatted text already in streamie?

Comment (cramforce) [inline public/ext/notification.js]: Please use 2 space indent everywhere

File: public/ext/notification.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    
+    if (i > last_i) {
+        result += escapeHTML(tweet.text.substring(last_i, i));
+    }
+    
+    return result;
+}
+
+function openLink(url)
+{
```
Comment (cramforce) [inline public/ext/notification.js]: Plase put opening brace on line of function statements

File: public/lib/stream/streamplugins.js [hunk 7]
```diff
@@ -443,29 +443,27 @@ require.def("stream/streamplugins",
...
-              if(!tweet.mentioned && !tweet.direct_message && !settings.get('notifications', 'tweets')) {
-                return
-              }
             try {
-              var notification = 
-                window.webkitNotifications.createNotification(tweet.data.user.profile_image_url, 
-                  tweet.data.user.name, 
-                  tweet.data.text);
-              notification.show();
+			  var tweetHash = {
```
Comment (cramforce) [inline public/lib/stream/streamplugins.js]: Why not just use the tweet datastructure directly?

You removed a lot of code. Did you want to do that? I think those checks make a lot of sense.

File: public/notification.html [hunk 8]
```diff
@@ -0,0 +1,92 @@
+<html>
```
Comment (cramforce) [inline public/notification.html]: Please put in sub directory together with JS file

**Comment by cramforce** [comment]
Thank You, looks great! Please see my inline comments

File: public/lib/stream/streamplugins.js (hunk 7 above)
Comment (ArdaXi) [inline public/lib/stream/streamplugins.js]: In this, I tried to be as brief as possible. I have no idea what the limit on window hashes is. This is all that I needed in the notification.
As for the checks, I don't think those were in there at the time I created this commit. I don't see why I would have removed them.

File: public/ext/notification.js (hunk 5 above)
Comment (ArdaXi) [inline public/ext/notification.js]: I wasn't sure whether that did the same thing, if it does, is it added to the tweet datastructure?

File: public/ext/jquery.timeago.js (hunk 2 above)
Comment (ArdaXi) [inline public/ext/jquery.timeago.js:1]: Either way should be possible, all this plugin needs is a properly formatted timestamp in the tag. I think Twitter itself uses this.

**Comment by ArdaXi** [comment]
Edited according to suggestions, making new request.

**Pull request closed without merging**